src/fn/edit-contacts.js

```diff
@@ -34,16 +52,29 @@ const execute = () => {
         const nameParts = fs.path.basename(csv).split('.');
         const prefix = nameParts[0];
         switch(prefix) {
-          case 'contact':  return processDocuments(csv, getIDs(csv, prefix), db, args);
-          case 'users': return processUsers(csv, getIDs(csv, prefix), db, args);
+          case 'contact':  return processDocuments('contact', csv, getIDs(csv, prefix), db, args);
+          case 'users': return processDocuments('user', csv, getIDs(csv, prefix), db, args);
           default: throw new Error(`Unrecognised CSV type ${prefix} for file ${csv}`);
         }
       })
       .then(docs => addToModel(csv, docs)),
       Promise.resolve())
 
     .then(() => model.exclusions.forEach(toDocs.removeExcludedField))
-    .then(() => Promise.all(Object.values(model.docs).map(saveJsonDoc)));
+    .then(() => Promise.all(Object.values(model.docs).map(doc => {
+      const saveDoc = saveJsonDoc(doc, args, overwriteAllFiles);
+
+      if(saveDoc === DOC_EXISTS) {
+        const userSelection = userPrompt.keyInSelect();
+
+        if (userSelection === undefined || userSelection === 2) {
+          throw new Error('User canceled the action.');
+        }
+        overwriteAllFiles = (userSelection === 1);
```

Comment:
Mutating a variable that is used with different conditions here and above, especially with async behavior in the mix, is pretty fragile, running the risk of bugs now or in the future.

Based on the logic, I get the sense there's a case above on line 65 where we want to use the default (ignoring user selection) and then this case here where we want to respect the user selection. With that, line 65 can just use a named constant (set to false) and then here on this line, a new const can be created to name the user selection; something like `const userForceOverwrite = (userSelection === 1);`

Reply:
One of the options a user gets is to overwrite all files which is a selection we need to keep track of and not prompt on subsequent files that are to be overwritten thus the current implementation.